src/raw/mod.rs

```diff
@@ -1408,6 +1422,126 @@ pub struct RawIter<T> {
+                // So the toggle isn't relevant to this iterator.
+                return;
+            }
+
+            if b.as_ptr() <= self.iter.data.next_n(Group::WIDTH).as_ptr() {
+                // The iterator has not yet reached the bucket's group.
+                // We don't need to reload anything, but we do need to adjust the item count.
+
+                if cfg!(debug_assertions) {
+                    // Double-check that the user isn't lying to us by checking the bucket state.
+                    // To do that, we need to find its control byte. We know that self.iter.data is
+                    // at self.iter.next_ctrl - Group::WIDTH, so we work from there:
+                    let offset = offset_from(self.iter.data.as_ptr(), b.as_ptr());
+                    let ctrl = self.iter.next_ctrl.sub(Group::WIDTH).add(offset);
+                    // This method should be called _before_ a removal, or _after_ an insert,
+                    // so in both cases the ctrl byte should indicate that the bucket is full.
+                    assert!(is_full(*ctrl));
+                }
+
+                if is_insert {
+                    self.items += 1;
+                } else {
+                    self.items -= 1;
+                }
+
+                return;
+            }
+
+            // The iterator is at the bucket group that the toggled bucket is in.
+            // We need to do two things:
+            //
+            //  - Determine if the iterator already yielded the toggled bucket.
+            //    If it did, we're done.
+            //  - Otherwise, update the iterator cached group so that it won't
+            //    yield a to-be-removed bucket, or _will_ yield a to-be-added bucket.
+            //    We'll also need ot update the item count accordingly.
+            if let Some(index) = self.iter.current_group.lowest_set_bit() {
+                let next_bucket = self.iter.data.next_n(index);
+                use core::cmp::Ordering;
+                match b.as_ptr().cmp(&next_bucket.as_ptr()) {
+                    Ordering::Greater => {
+                        // The toggled bucket is "before" the bucket the iterator would yield next.
+                        // We therefore don't need to do anything --- the iterator has already
+                        // passed the bucket in question.
+                        //
+                        // The item count must already be correct, since a removal or insert
+                        // "prior" to the iterator's position wouldn't affect the item count.
+                    }
+                    Ordering::Equal => {
+                        // The toggled bucket was coming up next, so we just skip it.
+                        self.iter.current_group = self.iter.current_group.remove_lowest_bit();
+                        // It must be a removal, since its bit was set.
+                        self.items -= 1;
+                        debug_assert!(!is_insert);
```

Comment:
It seems that this is just a simplified version of the case below? Why not just merge it with that case so we only need a single `<=` check.